stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt

```diff
@@ -116,6 +117,11 @@ public class ChannelListView : FrameLayout {
      */
     @JvmOverloads
     public fun setLoadingView(view: View, layoutParams: LayoutParams = defaultChildLayoutParams) {
+        // Make sure the loading view has no parent.
+        view.parent?.let { parent ->
+            (parent as ViewGroup).removeView(view)
+        }
```

Comment:
Good catch, we do need a check here.

Perhaps we could be more aggressive here and just throw an exception for this case. If clients write code where they pass in a View that has a parent, that's a clear bug. Us removing something randomly might be unexpected as well.

So something like

```kotlin
require(view.parent == null)
```

Also, we have custom empty views in addition to custom loading views, and we have this in other classes too. We should add checks everywhere.

Reply:
Yes, good idea. I will update with `require` instead of detaching from the parent.